kernel/aster-nix/src/fs/file_table.rs

```diff
@@ -122,7 +127,15 @@ impl FileTable {
             self.notify_fd_events(&events);
             entry.as_ref().unwrap().notify_fd_events(&events);
         }
-        entry.map(|e| e.file)
+
+        if let Some(file) = entry.map(|e| e.file) {
```

Comment:
There are essentially three separation checks for `entry.is_some()`.

The first one:

```rust
if entry.is_some() {
```

The second one

```rust
entry.map(|e| e.file)
```

And now the new third one:

```rust
if let Some(file) = 
```

The three checks should be combined into one by returning early.

```rust
    pub fn close_file(&mut self, fd: FileDesc) -> Option<Arc<dyn FileLike>> {
        let removed_entry = self.table.remove(fd as usize);
        let Some(closed_file) = removed_entry else {
            return None;
        }

        if let Some(closed_inode_file) = closed_file.downcast_ref::<InodeHandle>() {
            closed_inode_file.release_range_locks();
        }
        Some(closed_file)
```

Reply:
Thank you for your suggestion! After combining the `notify` logic, we have 
```rust
pub fn close_file(&mut self, fd: FileDesc) -> Option<Arc<dyn FileLike>> {
        if let Some(removed_entry) = self.table.remove(fd as usize) {
            let events = FdEvents::Close(fd);
            self.notify_fd_events(&events);
            removed_entry.notify_fd_events(&events);

            let closed_file = removed_entry.file;
            if let Some(closed_inode_file) = closed_file.downcast_ref::<InodeHandle>() {
                closed_inode_file.release_range_locks();
            }
            return Some(closed_file);
        }
        None
    }
```

Reply:
Do you know what I call the return-early-on-error technique? See a [recent explanation here](https://github.com/asterinas/asterinas/pull/1211#discussion_r1725182104). And after reading it, you will understand why the above code is not as readable as it can be.

Reply:
I think I have understood it now. And I simplify it by `?` as Rust Clippy instructed.  
```rust
    pub fn close_file(&mut self, fd: FileDesc) -> Option<Arc<dyn FileLike>> {
        let removed_entry = self.table.remove(fd as usize)?;

        let events = FdEvents::Close(fd);
        self.notify_fd_events(&events);
        removed_entry.notify_fd_events(&events);

        let closed_file = removed_entry.file;
        if let Some(closed_inode_file) = closed_file.downcast_ref::<InodeHandle>() {
            closed_inode_file.release_range_locks();
        }
        Some(closed_file)
    }
```